src/stan/callbacks/chained_writer.hpp

```diff
@@ -0,0 +1,89 @@
+#ifndef STAN_CALLBACKS_CHAINED_WRITER_HPP
+#define STAN_CALLBACKS_CHAINED_WRITER_HPP
+
+#include <stan/callbacks/writer.hpp>
+#include <ostream>
+#include <vector>
+#include <string>
+
+namespace stan {
+  namespace callbacks {
+
+    /**
+     * <code>chained_writer</code> is an implementation that writes to
+     * multiple writers.
+     *
+     * For any given call to this writer, it will delegate to both writers
+     * provided in the constructor.
+     */
+    class chained_writer : public writer {
```

Comment:
You're absolutely right. Any suggestions for names? `tee_writer` or `teed_writer`? `forked_writer`?